src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -117,21 +117,10 @@ private void Init(string pattern, RegexOptions options, TimeSpan matchTimeout, C
             // and is used as input into RegexOptions.Compiled and RegexOptions.NonBacktracking.
             _code = RegexWriter.Write(tree, culture);
 
-            if ((options & RegexOptions.NonBacktracking) != 0)
-            {
-                // NonBacktracking doesn't support captures (other than the implicit top-level capture).
-                capnames = null;
-                capslist = null;
-                caps = null;
-                capsize = 1;
-            }
-            else
-            {
```

Comment:
Even if just a few lines here and there, it's really nice to see special-casing like this go away. :smile: